modules/quest/src/main/java/unnamed/mmo/quest/Quest.java

```diff
@@ -0,0 +1,32 @@
+package unnamed.mmo.quest;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minestom.server.utils.NamespaceID;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import unnamed.mmo.quest.objective.Objective;
+import unnamed.mmo.registry.Registry;
+import unnamed.mmo.registry.Resource;
+import unnamed.mmo.util.ExtraCodecs;
+
+public record Quest(
+        @NotNull NamespaceID namespace,
+        //todo
+//        @NotNull List<Object> requirements,
```

Comment:
Some cleanup of comments necessary here

Reply:
I would like to add this system eventually, but I think I need to see a bit more the requirements of the system, eg how we will render potential quests and their requirements, how we will actually start them, etc.